extension/sigv4authextension/config.go

```diff
@@ -41,10 +42,18 @@ type AssumeRole struct {
 // compile time check that the Config struct satisfies the component.Config interface
 var _ component.Config = (*Config)(nil)
 
-// Validate checks that the configuration is valid.
```

Comment:
why did you remove that doc?

Reply:
Now that Validate() effectively isn't doing anything, I removed the comment because it wasn't entirely accurate - I can re-add it though, if this is just a consistency thing

Reply:
Validate is still needed looks like, since these errors can be returned:
```
	if err != nil {
		return fmt.Errorf("could not retrieve credential provider: %w", err)
	}
	if credsProvider == nil {
		return fmt.Errorf("credsProvider cannot be nil")
	}
```

My 0.02c: move the checks to a Validate method, and keep this unmarshal method to initialize the config properly.